lib/material/material_gui.flow

```diff
@@ -1468,4 +1472,200 @@ MOverlayT(margin : Transform<double>, radius : Transform<double>, style : Transf
 		\sz -> MGetFocusGroup(\parent -> TFrameT(margin, radius, fselectLift(style, \s -> mgraphicsStyle2tgraphicsStyle(parent, s)), sz)),
 		true
 	)
+}
+
+MSegmentedButtons(
+	items : [Pair<Pair<Maybe<MText>, Maybe<MIcon>>, Pair<[MButtonState], DynamicBehaviour<bool>>>],
+	multiSelect : bool,
+	iconsOnly : bool,
+	size : WidthHeight,
+	graphicsStyles : [MGraphicsStyle]
+) -> Material {
+	defaultBgColor = extractStruct(graphicsStyles, MThemedFill(MSecondaryColor())).color;
+	defaultStrokeColor = extractStruct(graphicsStyles, MThemedStroke(MSecondaryColor())).color;
+
+    minCellWidthIcons = 40.0;
+    minCellWidthText = 60.0;
+    charWidthMultiplier = 8.0;
+    textPadding = 32.0;
+    iconPadding = 24.0;
+    cellPadding = 16.0;
+    buttonContentMargin = 48.0;
+
+	totalWidth = size.width;
+	totalHeight = size.height;
+	borderRadius = totalHeight / 2.0; // 50%
+
+	// Calculate required width for each button based on content
+	requiredWidths = map(items, \item -> {
+		title = if (iconsOnly) "" else eitherFn(item.first.first, \text -> text.text, \ -> "");
+		hasIcon = isSome(item.first.second);
+		textWidth = i2d(strlen(title)) * charWidthMultiplier;
```

Comment:
It is dirt guess of the width. Width can vary depending  on font/size/other factors. You can use TextWidthInspector to inspect it dynamically or at least getTWordMetrics to make it statically.